admin/server/auth/claims.go

```diff
@@ -247,3 +190,112 @@ func unionProjectRoles(a *adminv1.ProjectPermissions, b *database.ProjectRole) *
 		ManageProjectMembers: a.ManageProjectMembers || b.ManageProjectMembers,
 	}
 }
+
+// organizationPermissionsUser is a Claims implementation for resolve the organization permissions for user
+func (c *authTokenClaims) organizationPermissionsUser(ctx context.Context, orgID string) *adminv1.OrganizationPermissions {
+	c.Lock()
+	defer c.Unlock()
+
+	if perm, ok := c.orgPermissionsCache[orgID]; ok {
+		return perm
+	}
```

Comment:
If you move this back into `OrganizationPermissions`, the caching will work for both service and user tokens